x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java

```diff
@@ -1004,4 +993,58 @@ static TaskType inferenceConfigToTaskType(InferenceConfig config) {
             return null;
         }
     }
+
+    // Iterates over the batch sending 1 request at a time to avoid
+    // filling the ml node inference queue.
+    class BatchIterator {
+        private final AtomicInteger index = new AtomicInteger();
```

Comment:
Why do you need an `AtomicInteger`? Is there any concurrency?

Reply:
For visibility so the threads see the up to date value. Probably not necessary as when the thread starts it would pick up the correct current value even if this was an integer